azure-keyvault/azure/keyvault/custom/key_vault_authentication.py

```diff
@@ -140,18 +141,47 @@ def auth_callack(server, resource, scope):
             self.keyvault_data_client = KeyVaultClient(KeyVaultAuthentication(auth_callack))
     """
 
-    def __init__(self, authorization_callback):
+    def __init__(self, authorization_callback=None, credentials=None):
         """
         Creates a new KeyVaultAuthentication instance used for authentication in the KeyVaultClient
         :param authorization_callback: A callback used to provide authentication credentials to the key vault data service.  
         This callback should take three str arguments: authorization uri, resource, and scope, and return 
         a tuple of (token type, access token).
+        :param credentials:: Credentials needed for the client to connect to Azure.
+        :type credentials: :mod:`A msrestazure Credentials
+         object<msrestazure.azure_active_directory>`
         """
-        super(KeyVaultAuthentication, self).__init__()
+        if not authorization_callback and not credentials:
+            raise ValueError("Either parameter 'authorization_callback' or parameter 'credentials' must not be None.")
```

Comment:
This seems like a double negative.  How about:

Either parameter 'authorization_callback' or parameter 'credentials' must be specified.